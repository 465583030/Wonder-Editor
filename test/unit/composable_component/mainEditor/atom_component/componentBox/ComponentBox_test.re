open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "componentBox",
    () => {
      let _buildComponentBoxComponent = (header,closable) =>
        ReactTestRenderer.create(
          <ComponentBox
            header
            closable
            gameObjectComponent=<div> (DomHelper.textEl("simulate div component")) </div>
          />
        );
      let _getFromArray = (array, index) => ArrayService.getNth(index, array);
      beforeEach(
        () => {
          TestTool.closeContractCheck();
        }
      );
      afterEach(() =>{
          TestTool.openContractCheck();
      });
      describe(
      "test component arguments",
      () => {
        test(
          "create can't close componentBox component",
          () => {
            _buildComponentBoxComponent("newBox", false) |> ReactTestTool.createSnapshotAndMatch;
          }
        );
        test(
          "create closable componentBox component",
          () => {
            _buildComponentBoxComponent("newBox", true) |> ReactTestTool.createSnapshotAndMatch;
          }
        );
       }
      );
      describe(
       "test workflow",
       () => {
            let _triggerClickTriangle = (domChildren) => {
              let headerDiv = _getFromArray(domChildren, 0);
              let triangleDiv = _getFromArray(headerDiv##children, 0);
              BaseEventTool.triggerClickEvent(triangleDiv)
            };
            test(
              "click once triangle to hide content component",
              () => {
                let component = _buildComponentBoxComponent("newBox", true);
                BaseEventTool.triggerComponentEvent(component, _triggerClickTriangle);
                component |>  ReactTestTool.createSnapshotAndMatch;

              }
            );
            test(
              "click twice triangle to show content component",
              () => {
                let component = _buildComponentBoxComponent("newBox", true);
                BaseEventTool.triggerComponentEvent(component, _triggerClickTriangle);
                BaseEventTool.triggerComponentEvent(component, _triggerClickTriangle);
                component |>  ReactTestTool.createSnapshotAndMatch;
              }
            );
          }
        );
    }
  );