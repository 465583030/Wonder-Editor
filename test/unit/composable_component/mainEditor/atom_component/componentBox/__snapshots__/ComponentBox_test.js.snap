// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentBox test component arguments create can't close componentBox component 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-bottom"
      />
    </div>
    <div
      className="header-title"
    >
      newBox
    </div>
  </div>
  <div>
    simulate div component
  </div>
</article>
`;

exports[`componentBox test component arguments create closable componentBox component 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-bottom"
      />
    </div>
    <div
      className="header-title"
    >
      newBox
    </div>
    <span
      className="header-close"
    >
      x
    </span>
  </div>
  <div>
    simulate div component
  </div>
</article>
`;

exports[`componentBox test workflow click once triangle to hide content component 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-right"
      />
    </div>
    <div
      className="header-title"
    >
      newBox
    </div>
    <span
      className="header-close"
    >
      x
    </span>
  </div>
</article>
`;

exports[`componentBox test workflow click twice triangle to show content component 1`] = `
<article
  className="componentBox-component"
>
  <div
    className="header"
  >
    <div
      className="header-triangle"
      onClick={[Function]}
    >
      <span
        className="triangle-bottom"
      />
    </div>
    <div
      className="header-title"
    >
      newBox
    </div>
    <span
      className="header-close"
    >
      x
    </span>
  </div>
  <div>
    simulate div component
  </div>
</article>
`;
